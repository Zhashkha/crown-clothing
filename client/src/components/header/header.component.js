import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { ReactComponent as Logo } from "../../assets/crown.svg";

import {
  HeaderContainer,
  LogoContainer,
  OptionsContainer,
  OptionContainerLink
} from "./header.styles";
import CartIcon from "../cart-icon/cart-icon.component";
import CartDropdown from "../cart-dropdown/cart-dropdown.component";
import { selectCurrentUser } from "../../redux/user/user.selectors";
import { selectCartHidden } from "../../redux/cart/cart.selectors";
import { signoutStart } from "../../redux/user/user.actions";

export const Header = ({ currentUser, hidden, signoutStart }) => (
  <HeaderContainer>
    <LogoContainer to="/">
      <Logo />
    </LogoContainer>
    <OptionsContainer>
      <OptionContainerLink to="/shop">SHOP</OptionContainerLink>
      {currentUser ? (
        <OptionContainerLink as="div" onClick={signoutStart}>
          SIGN OUT
        </OptionContainerLink>
      ) : (
        <OptionContainerLink to="/signin">SIGN IN</OptionContainerLink>
      )}
      <CartIcon />
    </OptionsContainer>
    {hidden ? null : <CartDropdown />}
  </HeaderContainer>
);

const mapStateToProps = createStructuredSelector({
  currentUser: selectCurrentUser,
  hidden: selectCartHidden
});

const mapDispatchToProps = (dispatch) => ({
  signoutStart: () => dispatch(signoutStart())
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
