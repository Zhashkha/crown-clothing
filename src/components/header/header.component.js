import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { ReactComponent as Logo } from "../../assets/crown.svg";

import {
  HeaderContainer,
  LogoContainer,
  OptionsContainer,
  OptionContainerLink,
} from "./header.styles";
import { auth } from "../../firebase/firebase.utils";
import CartIcon from "../cart-icon/cart-icon.component";
import CartDropdown from "../cart-dropdown/cart-dropdown.component";
import { selectCurrentUser } from "../../redux/user/user.selectors";
import { selectCartHidden } from "../../redux/cart/cart.selectors";

const Header = ({ currentUser, hidden }) => (
  <HeaderContainer>
    <LogoContainer to="/">
      <Logo />
    </LogoContainer>
    <OptionsContainer>
      <OptionContainerLink to="/shop">
        SHOP
      </OptionContainerLink>
      {currentUser ? (
        <OptionContainerLink as='div' onClick={() => auth.signOut()}>
          SIGN OUT
        </OptionContainerLink>
      ) : (
        <OptionContainerLink to="/signin">
          SIGN IN
        </OptionContainerLink>
      )}
      <CartIcon />
    </OptionsContainer>
    {hidden ? null : <CartDropdown />}
  </HeaderContainer>
);

const mapStateToProps = createStructuredSelector({
  currentUser: selectCurrentUser,
  hidden: selectCartHidden,
});

export default connect(mapStateToProps)(Header);
