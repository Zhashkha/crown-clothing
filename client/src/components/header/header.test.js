import React from "react";
import { shallow } from "enzyme";

import { Header } from "./header.component";
import CartDropdown from "../cart-dropdown/cart-dropdown.component";

describe("test Header", () => {
  let wrapper;
  let mockSignoutStart;

  beforeEach(() => {
    mockSignoutStart = jest.fn();

    const mockProps = {
      currentUser: { uid: 1 },
      hidden: false,
      signoutStart: mockSignoutStart
    };

    wrapper = shallow(<Header {...mockProps} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call signoutStart on OptionContainerLink click", () => {
    wrapper.find("OptionContainerLink").at(1).simulate("click");
    expect(mockSignoutStart).toHaveBeenCalled();
  });

  it("should render CartDropdown when hidden is false", () => {
    expect(wrapper.exists(CartDropdown)).toBe(true);
  });
});
