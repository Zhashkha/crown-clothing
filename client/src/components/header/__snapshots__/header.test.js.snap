// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Header should match snapshot 1`] = `ShallowWrapper {}`;
